.github/workflows/pkgci_shark_ai.yml

```
@@ -48,6 +48,18 @@ jobs:
         run: |
           echo "PY_VERSION_NO_DOT=$(echo ${{ matrix.version }} | tr -d '.')" >> $GITHUB_ENV
 
+      - name: Setup UV caching
+        run: |
+          CACHE_DIR="${GITHUB_WORKSPACE}/.uv-cache"
+          echo "UV_CACHE_DIR=${CACHE_DIR}" >> $GITHUB_ENV
+          mkdir -p "${CACHE_DIR}"
+
+      - name: Cache UV packages
+        uses: actions/cache@v3
```

match the version we use elsewhere in the project

```suggestion
        uses: actions/cache@6849a6489940f00c2f30c0fb92c6274307ccb58a # v4.1.2
```

(and pin to a hash, https://github.com/ossf/scorecard/blob/main/docs/checks.md#pinned-dependencies)

---

Fixed!